### DISCUSSION TITLE
ScriptReference/Animations.AnimatorControllerLayer-stateMachine

### DISCUSSION THREAD

**Discussion by giscus** [Developer-Notes-Extension/Unity-Notes#535] [Notes]
# ScriptReference/Animations.AnimatorControllerLayer-stateMachine

https://docs.unity3d.com/ScriptReference/Animations.AnimatorControllerLayer-stateMachine.html

**Comment by Flewd** [comment]
When looking at a Synced Layer this object will be empty. In order to get Animation Clip data from a synced layer you can use [AnimatorController.GetStateEffectiveMotion](https://docs.unity3d.com/ScriptReference/Animations.AnimatorController.GetStateEffectiveMotion.html) and pass in syncedLayerIndex.
